### PR TITLE
Fix __repr__ crashing when a member is uninitialized or deleted

### PR DESCRIPTION
**Pull request by epsy**
# Pull Request Check List

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](http://www.attrs.org/en/latest/contributing.html) at least once, it will save you unnecessary review cycles!

- [x] Added **tests** for changed code.
- [ ] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/master/tests/utils.py).
- [ ] Updated **documentation** for changed code.
- [ ] Documentation in `.rst` files is written using [semantic newlines](http://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/master/changelog.d).

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

**Comment by epsy** [comment]
I just realized this might be a good fit to add to hypothesis testing despite the wording of the checklist. I will look into it tonight. In the meantime please let me know if this is something you want to merge.

File: changelog.d/308.breaking.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+The ``__repr__`` set by ``attrs`` no longer produces an ``AttributeError`` when
```
Comment (hynek) [inline changelog.d/308.breaking.rst]: [Semantic newlines](http://rhodesmill.org/brandon/2012/one-sentence-per-line/) in documentation please.

File: tests/test_dunders.py [hunk 2]
```diff
@@ -200,6 +200,13 @@ class C(object):
 
         assert "C(_x=42)" == repr(i)
 
+    def test__repr_uninitialized_member(self):
```
Comment (hynek) [inline tests/test_dunders.py]: This has a superfluous underscore and needs a [test docstring](https://jml.io/pages/test-docstrings.html).

**Comment by epsy** [comment]
Thanks for the prompt reviews and sorry about the delay. I've addressed both comments.

File: changelog.d/308.breaking.rst (hunk 1 above)
Comment (wsanchez) [reply changelog.d/308.breaking.rst]: Huh, I'd never read that doc.  Cool.

Comment (wsanchez) [reply changelog.d/308.breaking.rst]: verified fixed

**Pull request merged**